### ISSUE TITLE
Threads die if they attempt to crawl a deleted directory

### ISSUE THREAD

**Issue by rtkennedy** [shirosaidev/diskover#11]
In crawlFiles I think there needs to be some error handling in cases where a directory or file has been deleted between when it was added to the crawl list and when it is recursed into or stat'd. If this happens for a directory, you get a traceback like this:

```
Exception in thread Thread-4:
Traceback (most recent call last):
  File "/usr/lib64/python2.7/threading.py", line 811, in __bootstrap_inner
    self.run()
  File "/usr/lib64/python2.7/threading.py", line 764, in run
    self.__target(*self.__args, **self.__kwargs)
  File "diskover.py", line 327, in processDirectoryWorker
    filelist = crawlFiles(path, DATEEPOCH, DAYSOLD, MINSIZE, EXCLUDED_FILES, VERBOSE)
  File "diskover.py", line 212, in crawlFiles
    for name in os.listdir(path):
OSError: [Errno 2] No such file or directory: '/no/longer/here'
```
And that thread is dead for the remainder of the scan. Since a directory or file can get yanked out from underneath a diskover thread at any time, is it possible to wrap the relevant bits of crawlFiles with a try/except blocks and have it gracefully move on in the event of an OSError? Or would respawning the thread be better/easier?

**Comment by shirosaidev** [comment]
Good catch, thanks. I've patched this in release **v1.0.10** with some other bug fixes and code improvements.

**Comment by rtkennedy** [comment]
This one is probably related to the patch (thanks for that, btw), it looks like it's calling printProgressBar with a dircount of 0?

```
Exception in thread Thread-4:
Traceback (most recent call last):
  File "/usr/lib64/python2.6/threading.py", line 532, in __bootstrap_inner
    self.run()
  File "/usr/lib64/python2.6/threading.py", line 484, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/var/tmp/diskover.py", line 335, in processDirectoryWorker
    printProgressBar(dircount, total_num_dirs, 'Crawling:', '%s/%s'%(dircount,total_num_dirs))
  File "/var/tmp/diskover.py", line 74, in printProgressBar
    percents = str_format.format(100 * (iteration / float(total)))
ZeroDivisionError: float division
```

**Comment by shirosaidev** [comment]
this has been fixed in **v1.0.11**
